internal/cli/root/atlas/builder.go

```diff
@@ -91,6 +123,10 @@ func Builder(profile *string) *cobra.Command {
 			"toc": "true",
 		},
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			handleSignal(cmd)
+
+			initProfile(profile)
```

Comment:
there are a few advantages to moving this code here:

1. easier to follow when is profile actually set (before any command runs
2. no need to pass pointers around (notice signature change in `func Builder`)
3. no need to use `cobra.OnInitialize` (it is a little misleading given initializers don't get triggered everytime)